File: Leetcoding-Actions/LC-Monthly-Challenges/2020-11/25-Smallest-Integer-Divisible-by-K.py
class Solution:
    def smallestRepunitDivByK(self, K: int) -> int:
        '''
        T: O(K) and S: O(1)
        '''
        remainder = 0
        
        for lengthN in range(1, K + 1):
            remainder = (remainder * 10 + 1) % K
            
            if remainder == 0:
                return lengthN
        
        return -1
            
            

File: Leetcoding-Actions/Explore-Monthly-Challenges/2020-11/25-Smallest-Integer-Divisible-by-K.java
class Solution {
    public int smallestRepunitDivByK(int K) {
        /*
        * T: O(K) and S: O(1)
        */
        int remainder = 0;
        
        for (int lengthN = 1; lengthN < K + 1; lengthN++) {
            remainder = (remainder * 10 + 1) % K;
            
            if (remainder == 0) {
                return lengthN;
            }
        }
        return -1;
    }
}
